File: finagle-example/src/main/scala/com/twitter/finagle/example/thrift/ThriftClient.scala
package com.twitter.finagle.example.thrift

import com.twitter.finagle.example.thriftscala.Hello
import com.twitter.finagle.Thrift
import com.twitter.util.Await

object ThriftClient {
  def main(args: Array[String]): Unit = {
    //#thriftclientapi
    val client = Thrift.client.build[Hello.MethodPerEndpoint]("localhost:8080")
    val response = client.hi().onSuccess { response => println("Received response: " + response) }
    Await.result(response)
    //#thriftclientapi
  }
}

File: finagle-example/src/main/java/com/twitter/finagle/example/java/thrift/ThriftClient.java
package com.twitter.finagle.example.java.thrift;

import scala.runtime.BoxedUnit;

import com.twitter.finagle.Thrift;
import com.twitter.finagle.example.thriftscala.Hello;
import com.twitter.util.Await;
import com.twitter.util.Function;
import com.twitter.util.Future;

public final class ThriftClient {

  private ThriftClient() { }

  /**
   * Runs the example with given {@code args}.
   *
   * @param args the argument list
   */
  public static void main(String[] args) throws Exception {
    //#thriftclientapi
    Hello.MethodPerEndpoint client =
        Thrift.client().build("localhost:8080", Hello.MethodPerEndpoint.class);
    Future<String> response = client.hi().onSuccess(new Function<String, BoxedUnit>() {
      @Override
      public BoxedUnit apply(String response) {
        System.out.println("Received response: " + response);
        return null;
      }
    });

    Await.result(response);
    //#thriftclientapi
  }
}
